aten/src/ATen/native/mkl/SpectralOps.cpp

```diff
@@ -46,104 +50,134 @@ namespace at { namespace native {
 // See NOTE [ Fourier Transform Conjugate Symmetry ] in native/SpectralOpsUtils.h.
 
 template <typename scalar_t>
-static inline void _fft_fill_with_conjugate_symmetry_slice(Tensor& output,
-                       int64_t signal_ndim, int64_t size_last_dim,
-                       int64_t start_last_dim_idx, int64_t i, int64_t num) {
-  scalar_t *data = output.data_ptr<scalar_t>();
-
-  // A slice means a slice of last dimension (of size size_last_dim)
-
-  // This function iterates through the slices to fill, i.e. to_slice_data
-  // (basically data_slices[i:i+num]), and keeps track of the slices it reads
-  // data from, i.e., from_slice_data, using from_slice_indices, a vector
-  // containing the index of the from_slice_data slice.
-
-  // Compute the indices for the first from_slice_data
-  std::vector<int64_t> from_slice_indices(signal_ndim);  // up to before last signal dim
-  int64_t remainder = i;
-  // set last signal dim values
-  int64_t from_slice_offset = 0;
-  for (int64_t d = signal_ndim - 1; d >= 0; d--) {
-    int64_t dim_size = output.size(d);
-    int64_t dim_idx = remainder % dim_size;
-    remainder = remainder / dim_size;
-    from_slice_indices[d] = dim_idx;
-    if (d == 0) {
-      from_slice_offset += dim_idx * output.stride(d);
-    } else if (dim_idx != 0) {
-      from_slice_offset += (dim_size - dim_idx) * output.stride(d);
-    }
-  }
+static __ubsan_ignore_undefined__  // UBSAN gives false positives on using negative indexes with a pointer
+void _fft_fill_with_conjugate_symmetry_slice(
+    Range range, at::ArrayRef<bool> is_mirrored_dim, IntArrayRef signal_half_sizes,
+    IntArrayRef in_strides, const scalar_t * in_ptr,
+    IntArrayRef out_strides, scalar_t * out_ptr) {
+  const int64_t ndim = signal_half_sizes.size();
```

Comment:
I think .size() here and in the for loop below returns an unsigned type so this should probably use auto